src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter df = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromNow = LocalDate.parse(dateFrom, df);
+        LocalDate dateToNow = LocalDate.parse(dateTo, df);
+        StringBuilder salaryBuilder = new StringBuilder();
+        salaryBuilder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            int count = 0;
+            for (String dataEntry : data) {
+                String[] separator = dataEntry.split(" ");
+                int hourlyRateIndex = Integer.parseInt(separator[3]);
```

Comment:
the same, not rename variable, just put 3 to constant with name HOURLY_RATE_INDEX